### PR TITLE
nwchem: init at 7.0.2

### PR DESCRIPTION
**Pull request by sheepforce**
###### Description of changes

Adds the NWChem quantum chemistry programme for large scale calculations on HPC systems. Part of the ongoing effort to upstream packages from NixOS-QChem, see https://github.com/markuskowa/NixOS-QChem/issues/146

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/chemistry/nwchem/default.nix [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  '';
+
+  passthru = { inherit mpi; };
+
+  meta = with lib; {
+    description = "Open Source High-Performance Computational Chemistry";
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ sheepforce markuskowa ];
+    homepage = "https://nwchemgit.github.io";
+    license = {
```
Comment (markuskowa) [inline pkgs/applications/science/chemistry/nwchem/default.nix:199]: That could now go into `lib/licenses.nix`. This license has an SPDX identifier: https://spdx.org/licenses/ECL-2.0.html

File: pkgs/applications/science/chemistry/nwchem/default.nix [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    amber_2 $out/share/nwchem/data/amber_q/
+    amber_3 $out/share/nwchem/data/amber_x/
+    amber_4 $out/share/nwchem/data/amber_u/
+    spce    $out/share/nwchem/data/solvents/spce.rst
+    charmm_s $out/share/nwchem/data/charmm_s/
+    charmm_x $out/share/nwchem/data/charmm_x/
+    EOF
+  '';
+
+  checkPhase = ''
```
Comment (markuskowa) [inline pkgs/applications/science/chemistry/nwchem/default.nix:171]: I think we should remove that. doCheck is set to false anyway.

**Review by markuskowa — APPROVED** [review]

File: pkgs/applications/science/chemistry/nwchem/default.nix [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    makeWrapper
+    gfortran
+  ];
+  buildInputs = [
+    tcsh
+    openssh
+    which
+    blas
+    lapack
+    which
```
Comment (wegank) [inline pkgs/applications/science/chemistry/nwchem/default.nix:70]:
```suggestion
```

File: pkgs/applications/science/chemistry/nwchem/default.nix [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    automake
+    autoconf
+    libtool
+    makeWrapper
+    gfortran
+  ];
+  buildInputs = [
+    tcsh
+    openssh
+    which
```
Comment (wegank) [inline pkgs/applications/science/chemistry/nwchem/default.nix:67]: Should `which` go to `nativeBuildInputs`?

Comment (sheepforce) [reply pkgs/applications/science/chemistry/nwchem/default.nix]: Oh yes, good spot!

**Pull request merged**